docs/deployment-ux.md

```diff
@@ -0,0 +1,410 @@
+to revision 1:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Inactive     0         0         0            0        
+SomeEnvironment/Prod:2    Active       5         5         5            5        
+```
+
+**Option 1**: Allow specifying earlier revisions to deploy. That
+revision becomes active again, and we deactivate the other revisions.
+
+``` {.shell}
+$ aws ecs deploy-environment --environment-name SomeEnvironment/Prod --environment-revision 1
+{ "DeploymentId": "SomeEnvironment/Prod:1@2017090611472301" }
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Deploying    0         5         5            5        
+SomeEnvironment/Prod:2    Undeploying  6         0         0            0        
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Active       5         5         5            5        
+SomeEnvironment/Prod:2    Reverted     0         0         0            0        
+```
+
+**Option 2**: Rollback copies the earlier revision to a new revision,
+and deploys that. The old revision remains inactive, and is not reused.
+
+``` {.shell}
+$ aws ecs deploy-environment --environment-name SomeEnvironment/Prod --environment-revision 1
+{ "DeploymentId": "SomeEnvironment/Prod:3@2017090611472301" }
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Inactive     0         0         0            0        
+SomeEnvironment/Prod:2    Undeploying  6         6         6            6        
+SomeEnvironment/Prod:3    Deploying    0         0         0            0        
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Inactive     0         0         0            0        
+SomeEnvironment/Prod:2    Inactive     0         0         0            0        
+SomeEnvironment/Prod:3    Active       6         6         6            6        
+```
+
+Pause ongoing deployments
+-------------------------
+
+Let's say halfway through a deployment, we realize that we made a
+mistake. Rolling back is one option, but might introduce more churn than
+we can handle. For this case, we can pause all deployments until we can
+fix the environment and deploy the right active revision.
+
+Open questions:
+
+-   Should we support a separate resume operation, or will we only
```

Comment:
To clarify, we'll be pausing Environments, not Deployments. In particular, pausing an environment means that it will not launch/terminate any tasks that belong to the environment until resumed.

We considered some scenarios where resume would be a useful separate concept to go along with pause:

- Something is wrong with the underlying cluster. We just want to pause the rollout until the underlying problem is fixed, and then resume where we left off. This could be particularly important for very large clusters.
- Some external large scale event is going on, and we don't want to continue making changes. For example: if we're doing a long slow rollout, but the ECS control plane or some startup dependency of the tasks we're launching becomes unavailable, then we want to pause cluster changes until the situation resolves. Neither rolling forward nor rolling back would make any progress, and could adversely affect fleet health. Once the situation resolves, we can resume from where we left off.

However, doing an explicit rollback/roll forward should implicitly resume the environment.

Reply:
On second thought, implicitly resuming could be dangerous if some automated system does a deployment while the reason for pausing the environment is still valid, that could cause harmful effects.